tests/durability/test_disk_failure_recovery.py

```diff
@@ -520,3 +521,460 @@ def test_sns_repair_fail_disk_diff_cvg_equal_to_k(self):
+                    near_full_percent)
+        resp = self.dsk_rec_obj.get_user_data_space_in_bytes(self.node_master_list[0],
+                                                             near_full_percent)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        if not resp[1]:
+            LOGGER.info("Current Memory usage is already more than expected memory usage,"
+                        " skipping write operation")
+        else:
+            workload_info = self.dsk_rec_obj.perform_near_full_sys_writes(s3userinfo=s3userinfo,
+                                                                          user_data_writes=resp[1],
+                                                                          bucket_prefix=
+                                                                          self.test_prefix[-1])
+
+        LOGGER.info("Step 2: Do IOs(Write and Read)")
+        self.test_prefix.append('test-36396')
+        self.s3_clean = users
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=s3userinfo,
+                                                    log_prefix=self.test_prefix[-1],
+                                                    skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("Step 3: Get degraded byte count before failing the disk")
+        degraded_byte_cnt_before = self.dsk_rec_obj.get_byte_count_hctl(self.hlth_master_list[0],
+                                                                        "degraded_byte_count")
+        LOGGER.info("degraded byte count: %s", degraded_byte_cnt_before)
+
+        LOGGER.info("Step 4: Fail disks less than K(parity units)")
+        LOGGER.info("No of parity units (K): %s", self.parity_units)
+        if self.parity_units == 1:
+            disk_fail_cnt = 1
+        else:
+            disk_fail_cnt = random.randint(1, self.parity_units - 1)  # nosec
+
+        resp = self.dsk_rec_obj.fail_disk(disk_fail_cnt, self.node_master_list[0],
+                                          self.node_worker_list, self.pod_name, on_diff_cvg=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.failed_disks_dict = resp[1]
+        time.sleep(self.delay_sns_repair)
+
+        LOGGER.info("Step 5: Get degraded byte count after disk failure")
+        degraded_byte_cnt_after_fail = self.dsk_rec_obj.get_byte_count_hctl \
+            (self.hlth_master_list[0], "degraded_byte_count")
+        LOGGER.info("degraded byte count: %s", degraded_byte_cnt_after_fail)
+
+        if degraded_byte_cnt_before >= degraded_byte_cnt_after_fail:
+            assert_utils.assert_true(False, "Degraded byte count after disk failure less than "
+                                            "or equal to degraded byte count before disk fail")
+        else:
+            LOGGER.info("Degraded byte count is more as expected after disk fail")
+
+        LOGGER.info("Step 6: Do IOs(Write and Read) after disk failure")
```

Comment:
are we checking anywhere that the system not going to read only as we are doing IOs on near full system?

Reply:
Maybe just add a TODO here then so we don't miss it.